libfwupdplugin/fu-device.c

```diff
@@ -718,6 +718,24 @@ fu_device_retry(FuDevice *self,
 	return fu_device_retry_full(self, func, count, priv->retry_delay, user_data, error);
 }
 
+/**
+ * fu_device_sleep:
+ * @self: a #FuDevice
+ * @delay_ms: delay in milliseconds
+ *
+ * Delays program execution, unless the device is emulated.
+ *
+ * Since: 1.8.11
+ **/
+void
+fu_device_sleep(FuDevice *self, guint delay_ms)
+{
+	g_return_if_fail(FU_IS_DEVICE(self));
+	g_return_if_fail(delay_ms < 1000);
```

Comment:
`fu_ti_tps6598x_device_wait_for_command()` is some times called with a 1000 ms delay.  
Should this be:
```suggestion
	g_return_if_fail(delay_ms <= 1000);
```
or the delays replaced by 999 in plugins/ti-tps6598x/fu-ti-tps6598x-device.c ?

Reply:
eek! I got my ms confused by my us! The idea there was to do a critical warning if the programmer tried to do `fu_device_sleep(100*1000)` i.e. in us. I think if we're waiting more than 100 seconds for something then it's probably better we use `fu_progress_sleep()` to get some progress indication. I've fixed up the constant and also added this in the docs. I said you had keen eyes! :)

Reply:
Okay fixed, this time also dealing with fu_progress_sleep too.